bug(test): clean app audit cannot resolve eliza-source TypeScript imports

## Problem

On a clean current-`develop` worktree, `bun run --cwd packages/app audit:app` fails before Playwright captures any view. The app renderer Vite build loads `@elizaos/cloud-routing` through the `eliza-source` export condition, then plain Node cannot resolve the ESM specifier `./features.js` in `src/index.ts` to the existing `src/features.ts`.

Observed failure:

```text
failed to load config from packages/app/vite.config.ts
Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'packages/cloud/routing/src/features.js' imported from packages/cloud/routing/src/index.ts
[ui-smoke] failed to start live stack: app renderer build failed
```

`packages/app/scripts/run-ui-playwright.mjs` appends `--conditions=eliza-source` to `NODE_OPTIONS` for source-only workspace imports, but it does not install the TypeScript resolver for child Node/Vite processes. Running the same required command with `NODE_OPTIONS='--import tsx'` reaches the live stack and begins capture.

## Acceptance criteria

- The unmodified `bun run --cwd packages/app audit:app` command starts from a clean frozen install without a caller-supplied `NODE_OPTIONS`.
- Source-condition `.js` specifiers resolve to workspace `.ts` files in the renderer build and Playwright collection.
- A regression test proves the runner preserves existing `NODE_OPTIONS` while adding both the source export condition and the required TypeScript resolver exactly once.
- The complete app audit passes and its screenshots/report are manually reviewed.

Parent: #13631

--- a/packages/app/scripts/lib/playwright-node-options.mjs
+++ b/packages/app/scripts/lib/playwright-node-options.mjs
@@ -1,0 +1,30 @@
+/**
+ * Normalizes Node options inherited by Playwright and its child build processes.
+ * Source-conditioned workspace exports point at TypeScript, so the export
+ * condition and tsx resolver must travel together through every child process.
+ */
+
+const SOURCE_CONDITION = "--conditions=eliza-source";
+const TSX_IMPORT = "tsx";
+
+export function withElizaSourceNodeOptions(value) {
+  const options =
+    typeof value === "string" && value.trim().length > 0
+      ? value.trim().split(/\s+/)
+      : [];
+
+  if (!options.includes(SOURCE_CONDITION)) {
+    options.push(SOURCE_CONDITION);
+  }
+
+  const hasTsxImport = options.some(
+    (option, index) =>
+      option === `--import=${TSX_IMPORT}` ||
+      (option === "--import" && options[index + 1] === TSX_IMPORT),
+  );
+  if (!hasTsxImport) {
+    options.push("--import", TSX_IMPORT);
+  }
+
+  return options.join(" ");
+}

--- a/packages/app/scripts/lib/playwright-node-options.test.mjs
+++ b/packages/app/scripts/lib/playwright-node-options.test.mjs
@@ -1,0 +1,31 @@
+/**
+ * Verifies that Playwright source-mode child processes inherit a stable,
+ * idempotent Node option set while preserving caller-supplied options.
+ */
+
+import { describe, expect, it } from "vitest";
+import { withElizaSourceNodeOptions } from "./playwright-node-options.mjs";
+
+describe("withElizaSourceNodeOptions", () => {
+  it("adds the source condition and TypeScript resolver", () => {
+    expect(withElizaSourceNodeOptions(undefined)).toBe(
+      "--conditions=eliza-source --import tsx",
+    );
+  });
+
+  it("preserves existing options and adds each requirement once", () => {
+    expect(withElizaSourceNodeOptions("--max-old-space-size=4096")).toBe(
+      "--max-old-space-size=4096 --conditions=eliza-source --import tsx",
+    );
+  });
+
+  it("is idempotent for the split import syntax", () => {
+    const options = "--trace-warnings --conditions=eliza-source --import tsx";
+    expect(withElizaSourceNodeOptions(options)).toBe(options);
+  });
+
+  it("recognizes the equals import syntax without adding a second loader", () => {
+    const options = "--import=tsx --conditions=eliza-source";
+    expect(withElizaSourceNodeOptions(options)).toBe(options);
+  });
+});

--- a/packages/app/scripts/run-ui-playwright.mjs
+++ b/packages/app/scripts/run-ui-playwright.mjs
@@ -8,6 +8,7 @@ import os from "node:os";
 import path from "node:path";
 import { fileURLToPath } from "node:url";
 import { getFreePort } from "../test/utils/get-free-port.mjs";
+import { withElizaSourceNodeOptions } from "./lib/playwright-node-options.mjs";
 
 const appDir = path.resolve(path.dirname(fileURLToPath(import.meta.url)), "..");
 const repoRoot = path.resolve(appDir, "..", "..");
@@ -180,17 +181,6 @@ function hasPlaywrightProject(projectName) {
   });
 }
 
-function appendNodeOption(value, option) {
-  const options =
-    typeof value === "string" && value.trim().length > 0
-      ? value.trim().split(/\s+/)
-      : [];
-  if (!options.includes(option)) {
-    options.push(option);
-  }
-  return options.join(" ");
-}
-
 function sleepSync(ms) {
   Atomics.wait(new Int32Array(new SharedArrayBuffer(4)), 0, 0, ms);
 }
@@ -349,14 +339,10 @@ async function getDistinctFreePort(excludedPorts = new Set()) {
 // Those packages publish an `eliza-source` export condition pointing at `src`;
 // on a fresh CI install (`bun install --ignore-scripts`) they have no `dist`, so
 // under default node conditions the collector resolves a missing
-// `dist/index.js` and the whole lane dies before any spec runs. The live-stack
-// web server (`run-node-tsx.mjs`) and the vitest source aliases already resolve
-// via `eliza-source`; the Playwright runner process must match so spec
-// collection reads the same source. No-op for packages without the condition.
-env.NODE_OPTIONS = appendNodeOption(
-  env.NODE_OPTIONS,
-  "--conditions=eliza-source",
-);
+// `dist/index.js` and the whole lane dies before any spec runs. The tsx import
+// is also required by child Node/Vite processes because source packages retain
+// NodeNext `.js` specifiers while their worktree files are TypeScript.
+env.NODE_OPTIONS = withElizaSourceNodeOptions(env.NODE_OPTIONS);
 
 if (hasPlaywrightConfig("playwright.ui-smoke.config.ts")) {
   env.ELIZA_UI_SMOKE_RUN_ID =

